src/coreComponents/mesh/generators/VTKUtilities.cpp

```diff
@@ -640,6 +640,115 @@ vtkSmartPointer< vtkDataSet > manageGlobalIds( vtkSmartPointer< vtkDataSet > mes
   return output;
 }
 
+/**
+ * @brief This function tries to make sure that no MPI rank is empty
+ *
+ * @param[in] mesh a vtk grid
+ * @param[in] comm the MPI communicator
+ * @return the vtk grid redistributed
+ */
+vtkSmartPointer< vtkDataSet >
+ensureNoEmptyRank( vtkDataSet & mesh,
+                   MPI_Comm const comm )
+{
+  // step 1: figure out who is a donor and who is a recipient
```

Comment:
I would maybe propose a caliper mark here so may later get an idea of how much this redistribution step contributes to the overall VTK mesh generation time
```suggestion
{
  GEOS_MARK_FUNCTION;

  // step 1: figure out who is a donor and who is a recipient
```

Reply:
Done